File: Case-4.py
print("Введите строку (IP-адрес).")
ip = input() #считываем IP-адрес
ip = ip.split('.') #разбивает строку на части, используя разделитель - '.'
if len(ip) == 4: #проверям, что строка разюбита на 4 части (если >/< => это уже не IP-адрес)
    if (ip[0].isdigit())and(ip[1].isdigit())and(ip[2].isdigit())and(ip[3].isdigit()): #проверка - число это/ есть посторонние символы
        #проверяем, что числа находятся в диапазоне от 0 до 255
        if (int(ip[0]) >= 0) and (int(ip[0]) <= 255) and (int(ip[1]) >=0 ) and (int(ip[1]) <= 255) and (int(ip[2]) >= 0) and (int(ip[2]) <= 255) and (int(ip[3]) >= 0) and (int(ip[3]) <= 255):
            print('YES')
        else:
             print('NO')
    else:
        print('NO')
else:
    print('NO')
input('Press ENTER to exit')

File: Goodline_Tasks_Swift/case4.swift
import Foundation
if CommandLine.arguments.count == 2 {

    let ip = CommandLine.arguments[1] 
    Calculation(stringIP:ip)   
} 
else
{
    if CommandLine.arguments.count > 1 
    {
        print("Ошибка ввода! Количество аргументов должно быть равно 1")
        exit(0)
    }
    
    let ip = readLine() //String/nil
    if ip == nil
    {
        print("Ошибка ввода!") 
        exit(0)
    }
    
    Calculation(stringIP:ip!)
}

func Calculation(stringIP: String)
{
    guard stringIP.first != "." && stringIP.last != "." else
    {print("NO"); exit(0)}

    let substring = stringIP.split(separator:".")
    guard substring.count == 4 else {print("NO"); exit(0)}

    for element in substring
    {
        guard let intElement: Int = Int(element) else 
        {print("NO"); exit(0)}

        guard intElement >= 0, intElement <= 255 else
        {print("NO"); exit(0)}
    }
    print("YES")
}